Update references of gradlePluginPortal to mavenCentral

mdedetrich opened the pull request:
The Gradle plugin now successfully deploys to maven central, see https://github.com/apache/incubator-pekko-grpc/actions/runs/5789438448/job/15690488703. The PR updates the documentation and also updates all references from `gradlePluginPortal`.

Resolves: https://github.com/apache/incubator-pekko-grpc/issues/113.

pjfanning commented:
There is a gradle.md that should be updated (paradox docs). The README links to the section in the gradle.md about the local build - so we should ensure that this link still works - or if we don't think that instructions on how to build the plugin should be in gradle.md, we can move the description to the readme.

mdedetrich commented:
> There is a gradle.md that should be updated (paradox docs). The README links to the section in the gradle.md about the local build - so we should ensure that this link still works - or if we don't think that instructions on how to build the plugin should be in gradle.md, we can move the description to the readme.

I just checked the file and indeed it seems that not even Lightbend/Akka updated it (it mentions that they haven't deployed to Gradle Plugin portal yet). Ill go ahead and update it so its up to scratch.

pjfanning commented:
> > There is a gradle.md that should be updated (paradox docs). The README links to the section in the gradle.md about the local build - so we should ensure that this link still works - or if we don't think that instructions on how to build the plugin should be in gradle.md, we can move the description to the readme.
> 
> I just checked the file and indeed it seems that not even Lightbend/Akka updated it (it mentions that they haven't deployed to Gradle Plugin portal yet). Ill go ahead and update it so its up to scratch.

I added the text about not supporting Gradle Plugin Repository in a previous commit.

mdedetrich commented:
> I added the text about not supporting Gradle Plugin Repository in a previous commit.

No worries, I just updated the text, let me know if its appropriate/

pjfanning commented on docs/src/main/paradox/buildtools/gradle.md:
````diff
@@ -13,25 +13,14 @@ By default, both client and server are generated and Java or Scala is autodetect
-or [download](https://pekko.apache.org/download.html) a source release, you can change directory to the
-`gradle-plugin` directory and run this Gradle command.
-
-```shell
````
Could you move these local build instructions to the README build from source section if you don't think we could keep them here?

mdedetrich replied:
Wouldn't `CONTRIBUTING.md` be a better spot for these instructions? As a user there really isn't a need to build the project manually, only if you are developing/doing changes to the plugin

pjfanning replied:
Maybe. It's just that it's useful to have a shortened version of the build instructions for release voters. At a minimum, we need to remove the Gradle link in the README because it will become a deal link.

mdedetrich replied:
Okay so I just added info to `README.md` and fixed the link, I think it should be fine now.

mdedetrich commented on .github/workflows/publish-nightly.yml:
```diff
@@ -31,7 +31,7 @@ jobs:
           NEXUS_USER: ${{ secrets.NEXUS_USER }}
           NEXUS_PW: ${{ secrets.NEXUS_PW }}
 
-      - name: Publish Gradle Plugin to Apache Nexus Repository
```
This extra info is irrelevant since entire plugin publishes to Apache Nexus repo

pjfanning reviewed (approved):
lgtm
